.travis.yml

```diff
@@ -2,31 +2,39 @@ language: generic
 
 sudo: required
 
-services:
-  - docker
+jobs:
+  include:
+    - language: python
+      name: Python tests
+      python: 3.6
+      services:
+        - docker
+      env:
+        - DOCKER_COMPOSE_VERSION=1.21.2
+      # install the latest version of docker
+      addons:
+        apt:
+          packages:
+            - docker-ce
 
-env:
-  - DOCKER_COMPOSE_VERSION=1.21.2
-
-before_install:
-  - sudo rm /usr/local/bin/docker-compose
-  - curl -L https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-`uname -s`-`uname -m` > docker-compose
-  - chmod +x docker-compose
-  - sudo mv docker-compose /usr/local/bin
-  - nvm install 10
-  - nvm use 10
-  # Fix https://github.com/travis-ci/travis-ci/issues/8982#issuecomment-354357640
-  - python -c "import fcntl; fcntl.fcntl(1, fcntl.F_SETFL, 0)"
-
-# install the latest version of docker
-addons:
-  apt:
-    packages:
-      - docker-ce
-
-script:
-  - make endtoendtest
-  - yarn run lint-all
-  # - yarn run lint:py
-
-cache: yarn
+      before_install:
+        - sudo rm /usr/local/bin/docker-compose
+        - curl -L https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-`uname -s`-`uname -m` > docker-compose
+        - chmod +x docker-compose
+        - sudo mv docker-compose /usr/local/bin
+        # Fix https://github.com/travis-ci/travis-ci/issues/8982#issuecomment-354357640
+        - python -c "import fcntl; fcntl.fcntl(1, fcntl.F_SETFL, 0)"
+      script:
+        - make docker-python-test
+    - language: node_js
```

Comment:
it reads like we don't need the docker env anymore for JS tests and linting. Is that right?

Reply:
Yep - we only needed that for running Python tests with the production backend. If we had full end to end tests, we would need it for the frontend too, but it seemed like unnecessary setup for the linting and frontend tests when they can run in Travis' own container.